torchgeo/datasets/geo.py

```diff
@@ -308,14 +307,16 @@ def files(self) -> list[Path]:
         # Using set to remove any duplicates if directories are overlapping
         files: set[Path] = set()
         for path in paths:
-            if os.path.isdir(path):
-                pathname = os.path.join(path, '**', self.filename_glob)
-                files |= set(glob.iglob(pathname, recursive=True))
-            elif (os.path.isfile(path) or path_is_vsi(path)) and fnmatch.fnmatch(
-                str(path), f'*{self.filename_glob}'
-            ):
-                files.add(path)
-            elif not hasattr(self, 'download'):
+            file_not_found = True
```

Comment:
What's the purpose of this variable? Can't we just use if-elif-else? There are no loops

Reply:
There are two cases where I currently warn:
1. it is a local file, but does not match the filename_glob.
2. It is a directory or a VFS, but no files matching filename_glob is found.

The first one is in nested `if`s so that if the inner if fails, it will not go the the outer elif. It should not go to the outer elif, at it is indeed a file.

I could duplicate the warning, but thought it was cleaner to have the warning defined once. 

Or we could remove the warning for case 1. as in this case the user has provided a path that is ignored?

Reply:
I did this because if it **is** a file, but does not match filename_glob, it does not need to list_directory. But I guess it is does not hurt to allow it to try listing the file.